File: setup_env.py
#!/usr/bin/env python

import shutil
import sys
import getopt

def main(argv):
    proxy_config_json = './proxy.config.json'
    proxy_config_json_skel = './proxy.config.json.skel'
    environment_ts = './src/environments/environment.ts'
    environment_ts_skel = './src/environments/environment.ts.skel'
    ip_addr = None
    helpmesg = "setup_env.py -i <ip_address>"

    try:
        opts, args = getopt.getopt(argv, "hi:", ["ip="])
    except getopt.GetoptError:
        print(helpmesg)
        sys.exit(2)
    if not opts:
        print(helpmesg)
        sys.exit(2)
    for opt, arg in opts:
        if opt == '-h':
            print(helpmesg)
            sys.exit(2)
        if opt in ('-i', '--ip'):
            ip_addr = arg

        shutil.copy(proxy_config_json_skel, proxy_config_json)
        shutil.copy(environment_ts_skel, environment_ts)


        with open(proxy_config_json) as proxy_file:
            new_proxy_conf = proxy_file.read().replace('$SERVER$', ip_addr)

        with open(proxy_config_json, "w") as proxy_file:
            proxy_file.write(new_proxy_conf)

        with open(environment_ts) as env_file:
            new_environment = env_file.read().replace('$SERVER$', ip_addr)

        with open(environment_ts, "w") as env_file:
            env_file.write(new_environment)

if __name__ == "__main__":
    main(sys.argv[1:])

File: setup_env.js
#!/usr/bin/env node

const program = require('commander');
const fs = require('fs');

program
  .option('-i, --ip <ip_address>', 'Sets IP address of your server')
  .parse(process.argv);

const proxyConfigJson = './proxy.config.json';
const environmentTs = './src/environments/environment.ts';
const url = normalizeUrl(program.ip);

printCurrentConfig();
console.log(''); // New line.

if (!url) {
  program.outputHelp();
  process.exit(2);
}

copySkel(proxyConfigJson);
copySkel(environmentTs);
console.log('Changing to: ' + url);

function normalizeUrl(url = '') {
  const parts = url.match(/^(?:https?:\/\/)?(?:[^@\n]+@)?([^:\/\n?]+)(?::([0-9]+))?/);

  if (!parts || !parts.length) {
    return;
  }

  let normalizedUrl = parts[1];
  if (parts[2]) {
    normalizedUrl = normalizedUrl + ':' + parts[2];
  }

  return normalizedUrl;
}

function copySkel(file) {
  const data = fs.readFileSync(file + '.skel', 'utf8');
  const result = data.replace(/\$SERVER\$/g, url);
  fs.writeFileSync(file, result, 'utf8');
}

function printCurrentConfig() {
  const doesConfigExist = fs.existsSync(proxyConfigJson);
  if (!doesConfigExist) {
    console.log('No current config set.');
    return;
  }
  const data = fs.readFileSync(environmentTs, 'utf8');
  const url = data.match(/remote: '([^']+)'/);
  console.log('Current server: ' + url[1]);
}
